plugins/forwarder/kafka/nativelog/forwarder.go

```diff
@@ -80,10 +82,16 @@ func (f *Forwarder) Forward(batch event.BatchEvents) error {
 		if !ok {
 			continue
 		}
-		for _, l := range data.LogList.Logs {
+		for _, logData := range data.LogList.Logs {
+			logdata := &v3.LogData{}
+			err := proto.Unmarshal(logData, logdata)
```

Comment:
BTW: Have you ever tested the CPU consumption of `Unmarshal` every time you transfer data? We found that the need for meaningless encoding and decoding in each transmission will consume a lot of CPU resources, so the byte array will be transmitted directly.

Reply:
Yes, there is indeed some impact on performance, although I haven't delved deeply into how much. In a previous version, I generated entities with only a few necessary fields for deserialization through a proto file. However, I found that I needed to maintain another entity. As a result, I deleted the generated files. Another ultimate solution, as you suggested, is to unpack the binary stream and obtain the corresponding fields based on the delimiter. We'll leave this as an optimization point for now, as it is sufficient for our current needs. However, as more applications are integrated later, we may optimize according to these solutions.

Reply:
I don't think we should loss performance due to add Kafka forward.
That isn't how things work. 
Please keep what we do for now, until open process mechanism discussion, and have new resolution.

Reply:
I agree with your point of view. Let's make it run first and optimize it later. Then, let's mark it as resolved.

Reply:
@mrproliu As a log forwarder, what it is the current way(gRPC) to process this? Is this consistent with that?

Reply:
When the current log uses gRPC to receive messages, it does not `Unmarshal`, it directly reads binary data for processing. When using gRPC to send data, directly send the current stream for ten days, because it does not need to do anything based on metadata, so he does not need to use `Umarshal` to do anything.
So, gRPC is fine.

Reply:
Then, we need to make Kafka follow the same way. If the gRPC forwards to the same channel, the Kafka forward should to the same partition.
@mrproliu Let's make sure this Kafka forward is always consistent with gRPC routing.

Reply:
For now, the gRPC didn't forward the same instance to the upstream, just make sure one stream send to the one upstream. 
If Kafka wants to forward to the same partition, then, just following what I say, decode the first log in the list, which is an element of the channel, don't decode them all.

Reply:
Why close it? this issue seems not been updated. please follow my comment to fix it.

Reply:
Ok, I known.